src/main/java/com/amazonaws/samples/beam/taxi/count/TaxiCount.java

```diff
@@ -78,87 +92,89 @@ public static void main(String[] args) {
       case "s3":
         batchSize = 20;
 
-        input = p
-            .apply("S3 source", TextIO
-                .read()
-                .from(options.getInputS3Pattern())
-            )
-            .apply("Parse S3 events",ParDo.of(new EventParser.S3Parser()));
+        input =
+            p.apply("S3 source", TextIO.read().from(options.getInputS3Pattern()))
+                .apply("Parse S3 events", ParDo.of(new EventParser.S3Parser()));
 
         LOG.info("Start consuming events from s3 bucket {}", options.getInputS3Pattern());
 
         break;
 
       default:
-        throw new IllegalArgumentException("expecting 'kinesis' or 's3' as parameter of 'inputSource'");
+        throw new IllegalArgumentException(
+            "expecting 'kinesis' or 's3' as parameter of 'inputSource'");
     }
 
-
-    PCollection<TripEvent> window = input
-        .apply("Group into 5 second windows", Window
-            .<TripEvent>into(FixedWindows.of(Duration.standardSeconds(5)))
-            .triggering(AfterWatermark
-                .pastEndOfWindow()
-                .withEarlyFirings(AfterProcessingTime.pastFirstElementInPane().plusDelayOf(Duration.standardSeconds(15)))
-            )
-            .withAllowedLateness(Duration.ZERO)
-            .discardingFiredPanes()
-        );
-
+    PCollection<TripEvent> window =
+        input.apply(
+            "Group into 5 second windows",
+            Window.<TripEvent>into(FixedWindows.of(Duration.standardSeconds(5)))
+                .triggering(
+                    AfterWatermark.pastEndOfWindow()
+                        .withEarlyFirings(
+                            AfterProcessingTime.pastFirstElementInPane()
+                                .plusDelayOf(Duration.standardSeconds(15))))
+                .withAllowedLateness(Duration.ZERO)
+                .discardingFiredPanes());
 
     PCollection<Metric> metrics;
 
-    if (! options.getOutputBoroughs()) {
-      metrics = window
-          .apply("Count globally", Combine
-              .globally(Count.<TripEvent>combineFn())
-              .withoutDefaults()
-          )
-          .apply("Map to Metric", ParDo.of(
-              new DoFn<Long, Metric>() {
-                @ProcessElement
-                public void process(ProcessContext c) {
-                  c.output(new Metric(c.element().longValue(), c.timestamp()));
-                }
-              }
-          ));
+    if (!options.getOutputBoroughs()) {
+      metrics =
+          window
+              .apply(
+                  "Count globally",
+                  Combine.globally(Count.<TripEvent>combineFn()).withoutDefaults())
+              .apply(
+                  "Map to Metric",
+                  ParDo.of(
+                      new DoFn<Long, Metric>() {
+                        @ProcessElement
```

Comment:
A static method or an overwriting the open method (if it exists) could potentially be used to initialize the profiler.